### PR TITLE
sg: add page

### PR DESCRIPTION
**Pull request by marcelarie**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** `0.13.0`

### CODE REVIEW

File: pages/common/sg.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# ast-grep
+
+> CLI tool for code structural search, lint, and rewriting.
```
[the remark this answers is not in the document]
Comment (marcelarie) [reply pages/common/sg.md:3]: haha true

File: pages/common/sg.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# ast-grep
+
+> Tool for code structural search, lint, and rewriting.
```
Comment (acuteenvy) [inline pages/common/sg.md:3]: The title of the page should be the command name.

The full name can then be put in the description:
https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#descriptions
```suggestion
# sg

> Ast-grep is a tool for code structural search, lint, and rewriting.
```

File: pages/common/sg.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+> More information: <https://ast-grep.github.io/guide/introduction.html>.
+
+- Scan possible queries on interactive mode:
+
+`sg scan --interactive`
+
+- Rewrite code using patterns on current directory:
```
Comment (acuteenvy) [inline pages/common/sg.md:10]:
```suggestion
- Rewrite code in the current directory using patterns:
```

File: pages/common/sg.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# ast-grep
+
+> Tool for code structural search, lint, and rewriting.
+> More information: <https://ast-grep.github.io/guide/introduction.html>.
+
+- Scan possible queries on interactive mode:
```
Comment (acuteenvy) [inline pages/common/sg.md:6]:
```suggestion
- Scan possible queries using interactive mode:
```

File: pages/common/sg.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# sg
+
+> Ast-grep is a tool for code structural search, lint, and rewriting.
+> More information: <https://ast-grep.github.io/guide/introduction.html>.
+
+- Scan possible queries using interactive mode:
```
Comment (acuteenvy) [inline pages/common/sg.md:6]:
```suggestion
- Scan for possible queries using interactive mode:
```

File: pages/common/sg.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+`sg run --pattern {{'foo'}} --rewrite {{'bar'}} --lang {{python}}`
+
+- Visualize possible changes without applying them:
+
+`sg -p 'useState<number>($A)' -r 'useState($A)' -l ts`
+
+- JSON Mode:
+
+`sg run -p 'Some($A)' -r 'None' --json | jq '.[].replacement' | jless`
```
Comment (acuteenvy) [inline pages/common/sg.md:20]:
```suggestion
- Output results as JSON, extract information using `jq` and interactively view it using `jless`:

`sg run --pattern '{{Some($A)}}' --rewrite '{{None}}' --json | jq '{{.[].replacement}}' | jless`
```

File: pages/common/sg.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+`sg scan --interactive`
+
+- Rewrite code in the current directory using patterns:
+
+`sg run --pattern {{'foo'}} --rewrite {{'bar'}} --lang {{python}}`
+
+- Visualize possible changes without applying them:
+
+`sg -p 'useState<number>($A)' -r 'useState($A)' -l ts`
```
Comment (acuteenvy) [inline pages/common/sg.md:16]: [long options](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax) and placeholders
```suggestion
- Visualize possible changes without applying them:

`sg run --pattern '{{useState<number>($A)}}' --rewrite '{{useState($A)}}' --lang {{typescript}}`
```

File: pages/common/sg.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+> Ast-grep is a tool for code structural search, lint, and rewriting.
+> More information: <https://ast-grep.github.io/guide/introduction.html>.
+
+- Scan possible queries using interactive mode:
+
+`sg scan --interactive`
+
+- Rewrite code in the current directory using patterns:
+
+`sg run --pattern {{'foo'}} --rewrite {{'bar'}} --lang {{python}}`
```
Comment (acuteenvy) [inline pages/common/sg.md:12]:
```suggestion
`sg run --pattern '{{foo}}' --rewrite '{{bar}}' --lang {{python}}`
```

**Review by gutjuri — APPROVED** [review]

**Review by acuteenvy — APPROVED** [review]

**Review by acuteenvy — APPROVED** [review]

**Pull request merged**